### ISSUE TITLE
Geometry column duplication 👻 mystery issue 👻 

### ISSUE THREAD

**Issue by anastasia** [ericrobskyhuntley/deduplicate-owners#12]
This issue has to do with as yet unpublished code, importing filings using `st_read` function
Here's the pertinent snippet:
```r
connect <- dbConnect(PostgreSQL(), 
                     dbname=local_db_name,
                     host="localhost",
                     port=5432)
query <- "select * from filings as f left join plaintiffs as p on p.docket_id = f.docket_id"

filings <- st_read(connect, query=query) %>% st_set_geometry("geometry") %>%
  modify_at("docket_id.1", ~NULL) %>% modify_at("id", ~NULL) %>% modify_at("id.1", ~NULL)
```

When running the `std_split_addresses` function on the `filings` table (`street` column), I get the error: 

`Can't transform a data frame with duplicate names.`
The problem is, before running the method, there were no duplicate column names in my filings table.
After running it, I have two `geometry` columns. Why?!

After commenting out most of the `std_split_addresses` function, I think the problem is happening at this extract function: https://github.com/ericrobskyhuntley/deduplicate-owners/blob/main/deduplicate-owners.R#L288

None of this makes any sense, and therefore seems to me that there might be a low-level bug somewhere. With lines 289 to 326 commented out, I am able to reproduce it consistently.

The current hacky solution is adding a "drop column" right after the extract function, hopefully that works.

**Comment by anastasia** [comment]
Both ```r
modify_at("geometry", ~NULL)``` and ```r  st_drop_geometry()``` cause the same error: ```Error in if (!setting_geom) { : missing value where TRUE/FALSE needed
```

Moving the drop column call past the mutate function results in the old error ```! Can't transform a data frame with duplicate names.
```

**Comment by ericrobskyhuntley** [comment]
Interesting - my best guess here is that you're dealing with the downstream impact of working with a spatial dataframe. I'm not going to have time to play with this today, but here's what I'd try...

First of  all, `st_drop_geometry()` should always be your go-to for getting rid of a geometry column (it changes the dataframe object type to a non-spatial df).

Second of all, it sounds like it's having trouble identifying the geometry column, which can cause some strange behavior. See what the geometry column is called when you read it in. (E.g., it might "geom", not "geometry"?) Then run `st_set_geometry("geometry_column _name") %>% st_set_geometry("geometry")` to set and rename. (See https://github.com/ericrobskyhuntley/deduplicate-owners/blob/main/deduplicate-owners.R#L798 ).

**Comment by ericrobskyhuntley** [comment]
Also, that `std_split_addresses()` function is by FAR the hackiest function in the script... check to make sure that none of your columns going in use the same names as the columns that it creates in separating the address and street fields.

**Comment by anastasia** [comment]
I added the hacky fix: 
```r
 %>% st_set_geometry("geometry") %>%
    st_drop_geometry("geometry") %>%
```
The column and its duplicate are both called `geometry`. Tried various drop/set combinations and this one seems to work. I chalk it up to R language's fault, as none of this makes much sense.

**Comment by ericrobskyhuntley** [comment]
I BELIEVE this should be fixed as of 261d39b ... see the (kind of unwieldy, but still helpful, `process_records()` function.
